File: hw/Vaca/u3/HW30ObserverAndStrategy/HW30StrategyJava/src/main/java/ec/edu/espe/view/SortView.java

package ec.edu.espe.view;

import java.util.Scanner;

/**
 *
 * @author MSI-PULSE
 */
public class SortView {
    public int[] getInputNumbers() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of elements: ");
        int n = scanner.nextInt();
        int[] numbers = new int[n];
        System.out.println("Enter the elements:");
        for (int i = 0; i < n; i++) {
            numbers[i] = scanner.nextInt();
        }
        return numbers;
    }

    public void displaySortedNumbers(int[] numbers, String strategy) {
        System.out.println("Sorted using " + strategy + ":");
        for (int number : numbers) {
            System.out.print(number + " ");
        }
        System.out.println();
    }
}

File: hw/Vaca/u3/HW30ObserverAndStrategy/HW30StrategyJavaScript/view/SortView.js
export class SortView {
    getInputNumbers() {
        const input = prompt("Enter the number of elements:");
        const n = parseInt(input);
        const numbers = [];

        for (let i = 0; i < n; i++) {
            const element = parseInt(prompt(`Enter element ${i + 1}:`));
            numbers.push(element);
        }

        return numbers;
    }

    displaySortedNumbers(numbers, strategy) {
        console.log(`Sorted using ${strategy}:`);
        console.log(numbers.join(" "));
    }
}
